### PR TITLE
[SecurityBundle][Security][Translation] trigger some deprecations for legacy methods

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Component/Translation/Writer/TranslationWriter.php [hunk 1]
```diff
@@ -41,7 +41,8 @@ public function addDumper($format, DumperInterface $dumper)
      */
```
Comment (chalasr) [inline src/Symfony/Component/Translation/Writer/TranslationWriter.php]: `@deprecated` annotations are missing

**Review by chalasr — COMMENTED** [review]
We also need to deprecate the `logout_on_user_change` firewall config node

File: src/Symfony/Component/Security/Http/Firewall/ContextListener.php [hunk 2]
```diff
@@ -70,7 +70,7 @@ public function __construct(TokenStorageInterface $tokenStorage, iterable $userP
      */
     public function setLogoutOnUserChange($logoutOnUserChange)
     {
-        // no-op, method to be deprecated in 4.1
+        @trigger_error(sprintf('The %s() method is deprecated since 4.1 and will be removed in 5.0.', __METHOD__));
```
Comment (hhamon) [inline src/Symfony/Component/Security/Http/Firewall/ContextListener.php]: missing `E_USER_DEPRECATED` error level.

File: src/Symfony/Component/Translation/Dumper/FileDumper.php [hunk 3]
```diff
@@ -49,12 +49,11 @@ public function setRelativePathTemplate($relativePathTemplate)
      */
     public function setBackup($backup)
     {
+        @trigger_error(sprintf('The %s() method is deprecated since 4.1 and will be removed in 5.0.', __METHOD__));
```
Comment (hhamon) [inline src/Symfony/Component/Translation/Dumper/FileDumper.php]: missing `E_USER_DEPRECATED` error level.

File: src/Symfony/Component/Translation/Writer/TranslationWriter.php [hunk 4]
```diff
@@ -41,7 +41,8 @@ public function addDumper($format, DumperInterface $dumper)
...
     public function disableBackup()
     {
-        // to be deprecated in 4.1
+        @trigger_error(sprintf('The %s() method is deprecated since 4.1 and will be removed in 5.0.', __METHOD__));
```
Comment (hhamon) [inline src/Symfony/Component/Translation/Writer/TranslationWriter.php]: missing `E_USER_DEPRECATED` error level.

**Comment by xabbuh** [comment]
comments addressed, tests pass

**Review by chalasr — APPROVED** [review]
👍

File: UPGRADE-4.1.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
+UPGRADE FROM 4.0 to 4.1
+=======================
+
+Security
+--------
+
+ * the `ContextListener::setLogoutOnUserChange()` method is deprecated and will be removed in 5.0
```
Comment (fabpot) [inline UPGRADE-4.1.md]: Missing cap and dot (same below) and in other files.

File: src/Symfony/Component/Translation/Writer/TranslationWriter.php (hunk 1 above)
Comment (fabpot) [reply src/Symfony/Component/Translation/Writer/TranslationWriter.php]: fixed

**Review by fabpot — CHANGES_REQUESTED** [review]

File: UPGRADE-4.1.md (hunk 5 above)
Comment (xabbuh) [reply UPGRADE-4.1.md]: fixed

**Review by fabpot — APPROVED** [review]

**Comment by chalasr** [comment]
Thank you @xabbuh.

**Pull request merged**